packages/cli/lib/services/deploy.service.ts

```diff
@@ -147,57 +147,77 @@ class DeployService {
+        const shouldConfirm = process.env['NANGO_DEPLOY_AUTO_CONFIRM'] !== 'true' && !autoConfirm;
+        const confirmationUrl = process.env['NANGO_HOSTPORT'] + `/sync/deploy/confirmation`;
+        try {
+            const bodyConfirmation: PostDeployConfirmation['Body'] = { ...postData, reconcile: false, debug, singleDeployMode };
+            const response = await http.post(confirmationUrl, bodyConfirmation, { headers: enrichHeaders() });
 
+            if (shouldConfirm) {
                 // Show response in term
                 console.log(JSON.stringify(response.data, null, 2));
+            }
 
-                const { newSyncs, deletedSyncs } = response.data;
+            const { newSyncs, deletedSyncs } = response.data;
 
-                for (const sync of newSyncs) {
-                    const actionMessage =
-                        sync.connections === 0 || sync.auto_start === false
-                            ? 'The sync will be added to your Nango instance if you deploy.'
-                            : `Nango will start syncing the corresponding data for ${sync.connections} existing connections.`;
-                    console.log(chalk.yellow(`Sync "${sync.name}" is new. ${actionMessage}`));
-                }
+            for (const sync of newSyncs) {
+                const syncMessage =
+                    sync.connections === 0 || sync.auto_start === false
+                        ? 'The sync will be added to your Nango instance if you deploy.'
+                        : `Nango will start syncing the corresponding data for ${sync.connections} existing connections.`;
+                console.log(chalk.yellow(`Sync "${sync.name}" is new. ${syncMessage}`));
+            }
 
-                for (const sync of deletedSyncs) {
-                    console.log(
-                        chalk.red(
-                            `Sync "${sync.name}" has been removed. It will stop running and the corresponding data will be deleted for ${sync.connections} existing connections.`
-                        )
-                    );
-                }
+            let deletedSyncsConnectionsCount = 0;
+            for (const sync of deletedSyncs) {
+                console.log(
+                    chalk.red(
+                        `Sync "${sync.name}" has been removed. It will stop running and the corresponding data will be deleted for ${sync.connections} existing connections.`
+                    )
+                );
+                deletedSyncsConnectionsCount += sync.connections;
+            }
 
-                const confirmation = await promptly.confirm('Do you want to continue y/n?');
+            // force confirmation :
+            // - if auto-confirm flag is not set
+            // - OR if there are deleted syncs with connections (and allow-destructive flag is not set)
+            const shouldConfirmDestructive = deletedSyncs.length > 0 && deletedSyncsConnectionsCount > 0 && !allowDestructive;
```

Comment:
Shouldn't it be
```suggestion
            const shouldConfirmDestructive = (deletedSyncs.length > 0 || deletedSyncsConnectionsCount > 0) && !allowDestructive;
```

Reply:
I was not planning to ask for confirmation if deleting a sync which has no connection since it is not deleting any records or anything that cannot be recoved.
True that `deletedSync.length > 0` is not necessary though since if this is the case would also be `0`. Will remove it